src/Gaffer/Animation.cpp

```diff
@@ -645,6 +661,33 @@ void Animation::Tangent::setSlopeWithScale( double slope, double scale, const bo
 		m_slope = slope;
 		m_scale = scale;
 	}
+
+	// tie slope and scale of opposite tangent
+	//
+	// NOTE : ensure that if both slope and scale of opposite tangent need to be tied we call
+	//        setSlopeWithScale() to limit the number of additional actions.
+
+	bool const tsl = m_key->tieSlopeActive();
+	bool const tsc = m_key->tieScaleActive() && ( m_key->m_tieScaleRatio != 0.0 );
+
+	if( tsl || tsc )
+	{
+		Tangent& ot = m_key->tangent( opposite( m_direction ) );
+		Private::ScopedAssignment< TieMode > tmGuard( m_key->m_tieMode, TieMode::Manual );
+
+		if( tsc )
+		{
+			const double oppositeScale = tieScaleOpposite( m_direction, scale, m_key->m_tieScaleRatio );
+
+			( tsl )
+				? ot.setSlopeWithScale( m_slope, oppositeScale, /* force = */ true )
+				: ot.setScale( oppositeScale, /* force = */ true );
+		}
+		else
+		{
+			ot.setSlope( m_slope, /* force = */ true );
+		}
```

Comment:
Are we relying on the equality check on line 615 to prevent the two tangents from ping-ponging back and forth setting each other in infinite recursion?  I think it probably works, but a comment might help understand what is actually happening.